filament/backend/src/CommandBufferQueue.cpp

```diff
@@ -56,6 +57,15 @@ void CommandBufferQueue::requestExit() {
     mCondition.notify_one();
 }
 
+void CommandBufferQueue::setPaused(bool paused) {
+    if (paused) {
+        mPaused = true;
+    } else {
+        mPaused = false;
+        mCondition.notify_one();
+    }
+}
```

Comment:
You need to take the lock while updating mPaused otherwise nothing guarantees that the change will be "seen" in the other thread (e.g. because of the caches).